eng/Directory.Build.Data.targets

```diff
@@ -80,7 +80,38 @@
   <PropertyGroup>
     <CentralPackagesFile>$(MSBuildThisFileDirectory)Packages.Data.props</CentralPackagesFile>
     <CentralPackageVersionPackagePath>$(MSBuildThisFileDirectory)Microsoft.Build.CentralPackageVersions\2.0.46\Sdk</CentralPackageVersionPackagePath>
+    <UseProjectReferenceToAzureClients Condition="'$(UseProjectReferenceToAzureClients)' == ''">false</UseProjectReferenceToAzureClients>
   </PropertyGroup>
 
+  <ItemGroup Condition="'$(UseProjectReferenceToAzureClients)' == 'true'">
+    <!-- Find all Azure projects -->
+    <AzureProjects Include="$(RepoRoot)/sdk/*/Azure.*/src/Azure.*.csproj" ProjectPath="%(Identity)" />
+    <PossibleProjectsToConvert Include="@(AzureProjects -> '%(Filename)')"/>
+
+    <!-- Compute set of projects that intersect with the PackageReference's -->
+    <_NotPackageReferences Include="@(PossibleProjectsToConvert)" Exclude="@(PackageReference)" />
+    <ProjectsToConvert Include="@(PossibleProjectsToConvert)" Exclude="@(_NotPackageReferences)" />
+
+    <!-- Remove PackageReferences -->
+    <PackageReference Remove="@(ProjectsToConvert)" />
+
+    <!-- Add ProjectReferences -->
+    <ProjectReference Include="@(ProjectsToConvert -> '%(ProjectPath)')" />
+  </ItemGroup>
+
+  <!-- Added layer of checks to make sure we correctly switched to project references -->
+  <Target Name="VerifyProjectReferencesReferences" AfterTargets="Build">
```

Comment:
@pakrym @jsquire  I added one extra level of checks that we verify all the Azure references are project references in this mode after a build completes which should help us ensure that they came from the sources instead of a package.